beacon-chain/sync/subscriber_blob_sidecar.go

```diff
@@ -4,29 +4,30 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed"
-	opfeed "github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed/operation"
-	eth "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
 	"google.golang.org/protobuf/proto"
 )
 
 func (s *Service) blobSubscriber(ctx context.Context, msg proto.Message) error {
-	b, ok := msg.(*eth.SignedBlobSidecar)
+	b, ok := msg.(blocks.VerifiedROBlob)
```

Comment:
I think this should be `ROBlob` instead of `VerifiedROBlob`. It's not verified yet right?

Reply:
I thought the way `subscribeStaticWithSubnets` works is:
- first calls `validator` (`type wrappedVal func(context.Context, peer.ID, *pubsub.Message) (pubsub.ValidationResult, error)`), which would be sync service's `validateBlob` in this instance
- validator sets the libp2p `Message.Validator` value to the result of performing validation
- if validation is successful, `handle` (`type subHandler func(context.Context, proto.Message) error`) gets called, and it responsible for casting `Message.Validator` to the type that the `validator` sets.

So I thought it makes sense for the validation function to be responsible for "upgrading" ROBlob->VerifiedROBlob once it passes validation. Then `handle` only gets called with the verified blob.

Reply:
My bad. I thought this was part of the validator handler. It's correct as it is now